nzedb/processing/Videos.php

```diff
@@ -84,15 +84,283 @@ protected function getSiteIDFromVideoID($siteColumn, $videoID)
 	 * @param string	$siteColumn
 	 * @param integer	$siteID
 	 *
-	 * @return array|false	False if invalid site, or ID not found; video.id value otherwise.
+	 * @return int|false	False if invalid site, or ID not found; video.id value otherwise.
 	 */
 	protected function getVideoIDFromSiteID($siteColumn, $siteID)
 	{
 		if (in_array($siteColumn, $this->sites)) {
 			$result = $this->pdo->queryOneRow("SELECT id FROM videos WHERE $siteColumn = $siteID");
 
-			return isset($result['id']) ? $result['id'] : false;
+			return isset($result['id']) ? (int)$result['id'] : false;
 		}
 		return false;
 	}
+
+	/**
+	 * Attempt a local lookup via the title first by exact match and then by like.
+	 * Returns a false for no match or the Video ID of the match.
+	 *
+	 * @param        $title
+	 * @param        $type
+	 * @param int    $source
+	 *
+	 * @return bool
+	 */
+	public function getByTitle($title, $type, $source = 0)
+	{
+		// Check if we already have an entry for this show.
+		$res = $this->getByTitleQuery($title, $type, $source);
+		if (isset($res['id'])) {
+			return $res['id'];
+		}
+
+		$title2 = str_replace(' and ', ' & ', $title);
+		if ($title != $title2) {
+			$res = $this->getByTitleQuery($title2, $type, $source);
+			if (isset($res['id'])) {
+				return $res['id'];
+			}
+			$pieces = explode(' ', $title2);
+			$title4 = '%';
+			foreach ($pieces as $piece) {
+				$title4 .= str_replace(["'", "!"], "", $piece) . '%';
+			}
+			$res = $this->getByTitleLikeQuery($title4, $type, $source);
+			if (isset($res['id'])) {
+				return $res['id'];
+			}
+		}
+
+		// Some words are spelled correctly 2 ways
+		// example theatre and theater
+		$title3 = str_replace('er', 're', $title);
```

Comment:
You misunderstand. $title2 is finished with. Use $title2 instead of $title3 to avoid a new variable use. Saves a small piece of RAM and is faster thancreating a brand new variable.